Q:

[SOLVED]ImportError: No module named pydot | rqt_graph

I'm running ROS Kinetic on Fedora 29.
I run these commands in the following way:
rosrun turtlesim turtlesim_node
rosrun turtlesim turtle_teleop_key

After these two commands I run rqt_graph and it gives the following error:
Warning: QT_DEVICE_PIXEL_RATIO is deprecated. Instead use:
   QT_AUTO_SCREEN_SCALE_FACTOR to enable platform plugin controlled per-screen factors.
   QT_SCREEN_SCALE_FACTORS to set per-screen factors.
   QT_SCALE_FACTOR to set the application global scale factor.
RosPluginProvider.load(rqt_graph/RosGraph) exception raised in __builtin__.__import__(rqt_graph.ros_graph, [RosGraph]):
Traceback (most recent call last):
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/rqt_gui/ros_plugin_provider.py", line 80, in load
    module = __builtin__.__import__(attributes['module_name'], fromlist=[attributes['class_from_class_type']], level=0)
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/rqt_graph/ros_graph.py", line 45, in <module>
    from qt_dotgraph.dot_to_qt import DotToQtGenerator
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/qt_dotgraph/dot_to_qt.py", line 36, in <module>
    import pydot
ImportError: No module named pydot

PluginManager._load_plugin() could not load plugin "rqt_graph/RosGraph":
Traceback (most recent call last):
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/qt_gui/plugin_handler.py", line 99, in load
    self._load()
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/qt_gui/plugin_handler_direct.py", line 54, in _load
    self._plugin = self._plugin_provider.load(self._instance_id.plugin_id, self._context)
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/qt_gui/composite_plugin_provider.py", line 71, in load
    instance = plugin_provider.load(plugin_id, plugin_context)
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/qt_gui/composite_plugin_provider.py", line 71, in load
    instance = plugin_provider.load(plugin_id, plugin_context)
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/rqt_gui_py/ros_py_plugin_provider.py", line 60, in load
    return super(RosPyPluginProvider, self).load(plugin_id, plugin_context)
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/qt_gui/composite_plugin_provider.py", line 71, in load
    instance = plugin_provider.load(plugin_id, plugin_context)
  File "/home/Giuseppe/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/rqt_gui/ros_plugin_provider.py", line 86, in load
    raise e
ImportError: No module named pydot

The initial warning doesn't seem to create much trouble.
I've made sure to install both python3-pydot and graphviz but nothing changed.

Originally posted by 0novanta on ROS Answers with karma: 18 on 2019-03-26
Post score: 0

A:

The problem was that I had installed python3-pydot but I needed python2-pydot.
Just doing sudo dnf install python2-pydot solved the issue.

Originally posted by 0novanta with karma: 18 on 2019-03-26
This answer was ACCEPTED on the original site
Post score: 0

